src/libraries/System.Private.CoreLib/src/System/IO/FileSystem.Unix.cs

```diff
@@ -372,23 +372,10 @@ public static void CreateDirectory(string fullPath)
 
         public static void MoveDirectory(string sourceFullPath, string destFullPath)
         {
-            // Windows doesn't care if you try and copy a file via "MoveDirectory"...
-            if (FileExists(sourceFullPath))
-            {
-                // ... but it doesn't like the source to have a trailing slash ...
-
-                // On Windows we end up with ERROR_INVALID_NAME, which is
-                // "The filename, directory name, or volume label syntax is incorrect."
-                //
-                // This surfaces as an IOException, if we let it go beyond here it would
-                // give DirectoryNotFound.
-
-                if (Path.EndsInDirectorySeparator(sourceFullPath))
-                    throw new IOException(SR.Format(SR.IO_PathNotFound_Path, sourceFullPath));
-
-                // ... but it doesn't care if the destination has a trailing separator.
+            // We have to check, if the source path does not end with a trailing separator.
+            // To match windows behaviour, Unix does not want trailing separators for both paths when it moves a file.
+            if (!Path.EndsInDirectorySeparator(sourceFullPath))
                 destFullPath = Path.TrimEndingDirectorySeparator(destFullPath);
-            }
```

Comment:
I'm not convinced about changing this behavior. This code was originally added in PR https://github.com/dotnet/corefx/pull/19716 to fix this issue https://github.com/dotnet/runtime/issues/21673 .

Reply:
If the code is untouched, we end up hitting FileExists twice on Unix. The checks will happen before and once we get to FileSystem.Unix.Move we saved a few round trips. But the behaviour should be the same as before

Reply:
Also, the change does not make sense: The new code is checking if _source path_ ends in separator, but then trimming the _destination path_.